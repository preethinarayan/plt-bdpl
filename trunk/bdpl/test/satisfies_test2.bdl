struct Outer
{
	byte fz (fieldsize 7) // this will contain the value 6 after reading file
		satisfies { 1 }
		then { print("fz found to be ");print(fz);print(".");}
		else { print("undefined fz!\n");};
	struct A
	{
		struct C
		{
			byte bb (fieldsize fz) // fieldsize = 6
			satisfies { 1 }
			then { print("bb is ");print(bb);print(".");}
			else { print("bb undefined ");}
			;
		}[fz]xx;
	
	}[*] a (fieldsize 79); // fieldsize is in bits !
}bb;



file "x:/test/file_test.dat" inp_file;
read(inp_file,bb);
