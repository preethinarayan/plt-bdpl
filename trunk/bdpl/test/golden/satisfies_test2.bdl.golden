fz found to be 0x6.bb is 0x28.bb is 0xa.bb is 0x4.bb is 0x21.bb is 0x28.bb is 0x22.bb is 0xa.bb is 0x23.bb is 0x8.bb is 0x3a.bb is 0x10.bb is 0x24.bb is 0x29.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.bb is 0x0.